apps/omg_utils/lib/omg_utils/http_rpc/client_adapter.ex

```diff
@@ -56,12 +58,44 @@ defmodule OMG.Watcher.HttpRPC.Adapter do
 
   def get_response_body(error), do: {:error, {:client_error, error}}
 
+  @doc """
+  Decodes specified keys in map from hex to binary
+  """
+  @spec decode16(map(), list()) :: map()
+  def decode16(data, keys) do
+    keys
+    |> Enum.into(%{}, &decode16_for_key(data, &1))
+    |> (&Map.merge(data, &1)).()
+  end
+
+  defp decode16_for_key(data, key) do
+    case data[key] do
+      value when is_binary(value) ->
+        {key, decode_binary!(value)}
+
+      value when is_list(value) ->
+        bin_list =
+          value
+          |> Enum.map(&Encoding.from_hex/1)
+          |> Enum.map(fn {:ok, bin} -> bin end)
+
+        {key, bin_list}
+    end
+  end
+
+  defp decode_binary!(value) do
```

Comment:
Name is suggesting it's something else than `decode16` but it's not...
Thinking about "matching" name: `decode16_value!` :question: